molten-trace/src/main/java/com/hotels/molten/trace/TracingTransformer.java

```diff
@@ -28,6 +28,8 @@
  * Transformer to wrap upstream in a trace span. The span is opened when upstream is subscribed to and closed when it completed, ran to an error, or was cancelled.
  * <p>
  * <strong>Be sure to have tracing initialized with {@link MoltenTrace#initialize()}.</strong>
+ * <strong>Please note that this solution not necessarily yields the expected result and should be handled with care.
+ * If the thread is changed between subscription and completion the span won't be nested and closed properly.</strong>
```

Comment:
Could you please create an issue for that?

Reply:
Created #36